tests/integration/integration_python_test.go

```diff
@@ -488,6 +488,76 @@ func TestCustomResourceTypeNameDynamicPython(t *testing.T) {
 	})
 }
 
+// Tests dynamic provider in Python with `serialize_as_secret_always` set to `False`.
+//
+//nolint:paralleltest // ProgramTest calls t.Parallel()
+func TestDynamicPythonDisableSerializationAsSecret(t *testing.T) {
+	dir := filepath.Join("dynamic", "python-disable-serialization-as-secret")
+	var randomVal string
+	integration.ProgramTest(t, &integration.ProgramTestOptions{
+		Dir: dir,
+		Dependencies: []string{
+			filepath.Join("..", "..", "sdk", "python", "env", "src"),
+		},
+		ExtraRuntimeValidation: func(t *testing.T, stack integration.RuntimeValidationStackInfo) {
+			randomVal = stack.Outputs["random_val"].(string)
+		},
+		EditDirs: []integration.EditDir{{
+			Dir:      filepath.Join(dir, "step1"),
```

Comment:
What s the empty edit here for? To check that nothing changes in a 2nd deployment?

Reply:
Yes, I believe so. The first deployment gets the random value output and the 2nd confirms it didn't change. I just based this off the existing test:

https://github.com/pulumi/pulumi/blob/598b9837b6cd2aef5000f0bb0890bc009225efb5/tests/integration/integration_python_acceptance_test.go#L52-L97